article/article.md

```diff
@@ -0,0 +1,153 @@
+
+## Flourishes
+
+By now, the main work was complete, but there were a few more nice-to-have features I was keen to add. First, I wanted to make the floating-point value of the pixel under the cursor available to user-defined mouse events. Since the parent class, `GridLayer`, maintains a cache of the tiles visible onscreen, doing so was pretty easy. I could use the cursor's screen position to determine (1) which tile was under the cursor and (2) the cursor's pixel coordinates within the tile. From there, the float value could be obtained from the binary pixel data with a JavaScript `DataView`:
+
+```typescript
+// create a DataView for obtaining a value from the binary pixel data
+const tileDataView = new DataView(pixelData.buffer);
+
+// To find the byte index:
+// (1) get the index of the start of the row in which the pixel is located
+//     (row index * number of columns)
+// (2) add to that the column index
+// (3) multiply by the number of bytes used for each pixel (4)
+const byteIndex = (coordsInTile.y * tileSize + coordsInTile.x) * BYTES_PER_WORD;
+
+// use the byte index and the machine's endianness to obtain the pixel value
+const pixelValue = tileDataView.getFloat32(byteIndex, littleEndian);
+```
+
+Second, I wanted to animate transitions (per pixel) when the user switched the view to load a new set of tiles. In addition to being a nice visual treat, I thought these transitions could be useful in the Local Burden of Disease visualization for illuminating changes over time. In particular, we typically provide a "play" control that allows users to view data for a series of years in a timed sequence. I found it difficult to perceive the magnitude of changes from year to year when viewing a sequence of static images, which was how we had originally implemented the play feature. I suspected, though, that animated transitions would help draw the eye to areas where more dramatic changes were occurring.
+
+![under-5 mortality 2000-2015 animation](./under5_mortality_animated.gif)
+
+To implement these transitions, I'd need to modify the WebGL renderer. Specifically, two textures would need to be loaded during transitions - one containing the old tiles and one containing the new. The shader code would need to compute the values of pixels in both the old and new tiles and interpolate between them over time. There were multiple cases to account for, because it was possible to (1) change the tiles but keep the colorization rules the same, and (2) change the tiles and the colorization rules. A third option would be to keep the same tiles but change the colorization rules. Since this wasn't possible in the Local Burden of Disease application, though, I decided to defer implementing it until it was actually needed.
```

Comment:
"Since this wasn't possible..." - Maybe make more clear that the third option is the one that isn't possible in LBD: in other words: what does "this" refer to?

Reply:
Changed the language in an attempt to clarify.